gammapy/modeling/models/temporal.py

```diff
@@ -472,79 +471,93 @@ def __str__(self):
-    def _time_ref(self):
-        return time_ref_from_dict(self.table.meta)
-
-    @lazyproperty
-    def _time(self):
-        return self._time_ref + self.table["TIME"].data * getattr(
-            u, self.table.meta["TIMEUNIT"]
-        )
+        time_axis = MapAxis.from_nodes(nodes=table["TIME"] + table.meta["MJDREFI"],
+                                       name="time", unit=table.meta["TIMEUNIT"])
+        axes = [time_axis]
 
-    def evaluate(self, time, ext=0):
-        """Evaluate for a given time.
+        if "ENERGY" in columns:
+            energy_axis = MapAxis.from_energy_bounds(table["ENERGY"], name="energy",
+                                                     )
+            axes.append(energy_axis)
+        m = RegionNDMap.create(region=None, axes=axes, meta=table.meta)
 
-        Parameters
-        ----------
-        time : array_like
-            Time since the ``reference`` time.
-        ext : int or str, optional, default: 0
-            Parameter passed to ~scipy.interpolate.InterpolatedUnivariateSpline
-            Controls the extrapolation mode for GTIs outside the range
-            0 or "extrapolate", return the extrapolated value.
-            1 or "zeros", return 0
-            2 or "raise", raise a ValueError
-            3 or "const", return the boundary value.
+        return cls(m, filename=filename, t_ref=t_ref)
 
+    @classmethod
+    def read(cls, filename, format="table"):
+        """Read a TemplateModel from disk
 
-        Returns
-        -------
-        norm : array_like
-            Norm at the given times.
-        """
-        return self._interpolator(time, ext=ext)
+        Parameters:
 
-    def integral(self, t_min, t_max):
-        """Evaluate the integrated flux within the given time intervals
+        filename : str
+            Name of file to read
+        format : str
+            Format of the input file.
+            either "table" or "map"
 
-        Parameters
-        ----------
-        t_min: `~astropy.time.Time`
-            Start times of observation
-        t_max: `~astropy.time.Time`
-            Stop times of observation
-        Returns
-        -------
-        norm: The model integrated flux
         """
+        filename = str(make_path(filename))
+        if format == "table":
+            table = Table.read(filename)
+            cls.from_table(table)
 
-        n1 = self._interpolator.antiderivative()(t_max.mjd)
-        n2 = self._interpolator.antiderivative()(t_min.mjd)
-        return u.Quantity(n1 - n2, "day") / self.time_sum(t_min, t_max)
+        elif format == "map":
+            m = RegionNDMap.read(filenames)
+            t_ref = m.meta["MJDREFI"]
+            return cls(m, filename=filename, t_ref=t_ref)
+
+        else:
+            raise ValueError("Not a valid format")
+
+    def to_table(self):
+        """Convert model to an astropy table"""
+        columns = self.map.geom.axes
+        table = Table(data=self.map.quantity, names=columns,
```

Comment:
This does not work for the extra axes. For now it is sufficient to just support the "old" format and raise an error for a 2d model instead. I think we should introduce the `format` argument here as well, even if only one format is supported for now.